addons/website_sale/models/product.py

```diff
@@ -183,14 +187,21 @@ def check_parent_id(self):
     def name_get(self):
         res = []
         for category in self:
-            names = [category.name]
-            parent_category = category.parent_id
-            while parent_category:
-                names.append(parent_category.name)
-                parent_category = parent_category.parent_id
-            res.append((category.id, ' / '.join(reversed(names))))
+            name = category.name
+            parent_name = category.parent_id.display_name
```

Comment:
While this is working on frontend it looks like it is not working in backend, `display_name` of `parent_id` will always be `False`.

That's an issue since we show `display_name` on the category list view to show hierarchy (eg `Cat1 / Cat2 / Cat3`), no?

Reply:
This worked for me locally but apparently not anymore. I think it's related to the id of the parent compared to the children or something like that, and I was in a favorable case by luck.

Since this is not reliable, I force pushed another efficient way to code the method, based on `parent_path` through `parents_and_self`.